cli/azd/pkg/project/container_helper.go

```diff
@@ -198,7 +200,21 @@ func (ch *ContainerHelper) Credentials(
 		return nil, err
 	}
 
-	return ch.containerRegistryService.Credentials(ctx, targetResource.SubscriptionId(), loginServer)
+	var credential *azcli.DockerCredentials
+	credentialsError := retry.Do(
+		ctx,
+		retry.WithMaxDuration(90*time.Second, retry.NewExponential(1*time.Second)),
+		func(ctx context.Context) error {
+			cred, err := ch.containerRegistryService.Credentials(ctx, targetResource.SubscriptionId(), loginServer)
```

Comment:
Is there a way we can implement this at the lower level where we speak to external processes? Generally, higher level application-layer retries are susceptible to accidental compounding retry loops.

Reply:
sdk clients provide a default retry logic, but the error 404 is usually not-retriable.

I wouldn't set a 404 error as a general-case to retry, as in most cases, it will means there's an error with the url used to make a request.

For this particular case, the `Credentials()` method is just an interface, so, adding the retry on top of it covers any implementation for it.

I think I can improve this code by retrying exclusively for 404 error

Reply:
I see. Thanks for the explanation, that makes a lot of sense. Adding the 404 catch for `POST <acr>.io/oauth/exchange` seems like a good change.

Reply:
changed applied to retry only for 404